File: 30days/day08-dictionaries-and-maps/Solution.js
// https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem

function processData(input) {
    //Enter your code here
    input = input.split('\n');

    let n = parseInt(input[0])
    let phoneBook = new Map(input.slice(1, n + 1).map(l => l.split(' ')));

    let contact;
    for (let i = n + 1; i < input.length; i++) {
        if (phoneBook.has(input[i])) {
            console.log(`${input[i]}=${phoneBook.get(input[i])}`);
        }
        else {
            console.log('Not found');
        }
    }
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});

File: 30days/day08-dictionaries-and-maps/Solution.cs
// https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem

using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static void Main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        int n = Convert.ToInt32(Console.ReadLine());
        Dictionary<string, string> phoneBook = new Dictionary<string, string>();
        var consoleLine = string.Empty;

        for(int i = 1; i <= n; i++) {
            consoleLine = Console.ReadLine();
            phoneBook.Add(consoleLine.Split(' ')[0], consoleLine.Split(' ')[1]);
        }
        
        consoleLine = Console.ReadLine();
        while (consoleLine != null) {
            if (phoneBook.ContainsKey(consoleLine)) {
                Console.WriteLine($"{consoleLine}={phoneBook[consoleLine]}");
            }
            else {
                Console.WriteLine("Not found");
            }
            consoleLine = Console.ReadLine();
        }
    }
}
